client/my-sites/plan-features-2023-grid/free-plan-paid-domain-dialog.tsx

```diff
@@ -0,0 +1,227 @@
+import { PlanSlug, getPlan } from '@automattic/calypso-products';
+import { Button, Dialog } from '@automattic/components';
+import { formatCurrency } from '@automattic/format-currency';
+import { Global, css } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import { useQueryClient } from 'react-query';
+import { useSelector } from 'react-redux';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import usePlanPrices from '../plans/hooks/use-plan-prices';
+import { LoadingPlaceHolder } from './components/loading-placeholder';
+import {
+	USE_GET_WORDPRESS_SUBDOMAIN_QUERY_KEY,
+	useGetWordPressSubdomain,
+} from './hooks/use-get-wordpress-subdomain';
+import type { DomainSuggestion } from '@automattic/data-stores';
```

Comment:
This import is what triggered my curiosity here. 

Are the domain suggestions fetched via the react-query hook something that we can already retrieve via the [getDomainSuggestions](https://github.com/Automattic/wp-calypso/blob/95a95c9c2825abf379695395792aaff51b0731d6/packages/data-stores/src/domain-suggestions/selectors.ts#L26) selector?

Reply:
We can use the selector, although we would have to write a wrapper around it that would expose error and loading states also.

Reply:
> expose error and loading states also.

These should be available via the core selectors that every store gets (`isResolving`/`hasFinishedResolution` should serve for loading states). you can see example usage in the [plans store PR](https://github.com/Automattic/wp-calypso/pull/76024/files#diff-e268185bf1c246017f6796f5f74ff3c0ae49b12e2acf525d8074859449573a19R1236). dunno about error states though.

In this case, you'd have something like (pseudo):
```
useSelect( ( select ) => ( {
    suggestions: select( STORE ).getDomainSuggestions( query ),
    resolved: select( STORE ).hasFinishedResolution( 'getDomainSuggestions', query ),
} ) )
```

Reply:
TIL! Thanks, will do this when I pick up the refactor.